test/zdtm/static/cpu-affinity0.c

```diff
@@ -0,0 +1,42 @@
+#include <errno.h>
+#include <stdlib.h>
+#include <sched.h>
+
+#include "zdtmtst.h"
+
+const char *test_doc	= "Check that with-cpu-affinity option can restore cpu affinity";
+const char *test_author	= "Sang Yan <sangyan@huawei.com>";
+
+int main(int argc, char **argv)
+{
+	cpu_set_t old;
+	cpu_set_t new;
+
+	test_init(argc, argv);
+
+	CPU_ZERO(&old);
+	CPU_ZERO(&new);
+
+	/* test only 0 core because of CI test env limited */
+	CPU_SET(0, &old);
+
+	if (sched_setaffinity(getpid(), sizeof(old), &old) < 0) {
+		pr_perror("Can't set old cpu affinity! errno: %d", errno);
```

Comment:
`pr_perror()` is defined in `test/zdtm/lib/zdtmtst.h` as
```C
#define pr_perror(format, arg...)       \
        test_msg("ERR: %s:%d: " format " (errno = %d (%s))\n", \
                __FILE__, __LINE__, ## arg, errno, strerror(errno))
```

Note that it already includes information about errno.